api/src/main/java/com/nicknathanjustin/streamercontracts/users/UsersApiController.java

```diff
@@ -24,8 +23,12 @@
     @NonNull private final UserService userService;
 
     @RequestMapping(path = "user", method = RequestMethod.GET)
-    public Principal user(@Nullable final Principal principal) {
-        return principal;
+    public User user(@NonNull final OAuth2Authentication auth) {
```

Comment:
Sir! Respect the @Nullable annotation!

The field can be null. And since we know the field can be null, we'll have to include a null check in the method